### PR TITLE
iReceptor importer

### PR DESCRIPTION
**Pull request by BlueBasilisk**
added a new iReceptor importer. This importer takes in a path with one or more .zip files as retrieved from the iReceptor gateway (zip files contain AIRR .tsv files and metadata .json files), converts this to conventional immuneML AIRR files (one file per repertoire) + metadata.csv file, and then uses AIRRImport to import this dataset. 
Also some minor updates in AIRRImport docs

### CODE REVIEW

**Review by pavlovicmilena — APPROVED** [review]
Looks good! I only had a few minor questions :)

File: test/IO/dataset_import/test_iReceptorImport.py [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+        self.assertListEqual(list(metadata_df["age_max"]), [80 for i in range(6)])
+        self.assertListEqual(list(metadata_df["age_event"]), ["Sample Collection" for i in range(6)])
+        self.assertListEqual(list(metadata_df["tissue_label"]), ["peripheral blood" for i in range(6)])
+        self.assertListEqual(list(metadata_df["collection_time_point_relative"]), ["More than 14 days" for i in range(6)])
+        self.assertListEqual(list(metadata_df["collection_time_point_reference"]), ["First negative COVID-19 test" for i in range(6)])
+        self.assertListEqual(list(metadata_df["COVID-19"]), ["Case", "Control", "Control", "Case", "Control", "Control"])
+        self.assertListEqual(list(metadata_df["second_disease"]), [np.nan, "Case", "Case", np.nan, np.nan, np.nan])
+        self.assertListEqual(list(metadata_df["first_disease"]), [np.nan,  np.nan, np.nan, np.nan, "Case", "Case"])
+
+        shutil.rmtree(path)
```
Comment (pavlovicmilena) [inline test/IO/dataset_import/test_iReceptorImport.py:473]: btw, ireceptorimport folder is not deleted here, only subfolders :)

File: source/IO/dataset_import/IReceptorImport.py [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+        any other value will result in importing the sequences as they are.
+        Valid values for region_type are the names of the :py:obj:`~source.data_model.receptor.RegionType.RegionType` enum.
+
+        column_mapping (dict): A mapping from AIRR column names to immuneML's internal data representation.
+        For AIRR, this is by default set to:
+
+        .. indent with spaces
+        .. code-block:: yaml
+
+                junction: sequences
```
[the remark this answers is not in the document]
Comment (BlueBasilisk) [reply source/IO/dataset_import/IReceptorImport.py:85]: All of this stuff is just the same as AIRR; junction is a required field (although nullable) whereas cdr3 is optional. However, IReceptorImport uses AIRRImport internally, which will use the cdr3 field if junction is null (e.g., immuneML-exported AIRR files)

File: source/IO/dataset_import/IReceptorImport.py [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+            with zipfile.ZipFile(zip_filename, "r") as zip_object:
+                for file in zip_object.filelist:
+                    file.filename = f"{Path(zip_filename).stem}_{file.filename}"
+                    if not file.filename.endswith("info.txt"):
+                        if unzip_metadata or not file.filename.endswith("-metadata.json"):
+                            zip_object.extract(file, path=unzipped_path)
+
+    @staticmethod
+    def _get_metadata_row(repertoire, sample, data_processing):
+        repertoire_id = repertoire['repertoire_id']
```
[the remark this answers is not in the document]
Comment (BlueBasilisk) [reply source/IO/dataset_import/IReceptorImport.py:216]: these are always fixed, these metadata json files follow an official AIRR format as well

**Review by pavlovicmilena — APPROVED** [review]

**Comment by pavlovicmilena** [comment]
Great, thanks! 🎉

**Pull request merged**